clang/include/clang/Driver/Job.h

```diff
@@ -48,6 +53,10 @@ class Command {
   /// The executable to run.
   const char *Executable;
 
+  /// The container for custom driver-set diagnostic messages that are
+  /// produced upon particular error codes returned by the command
+  ErrorCodeDiagMapTy ErrorCodeDiagMap;
```

Comment:
Buuut, as I can see the `llvm-no-spir-kernel` is not returning different non-null error codes in your latest patch... So, what's the purpose of these changes now? Do many tools return different non-null error codes?

Reply:
One of the ideas behind this infrastructure is to make it re-usable for any external tool, be it inside the SYCL toolchain or any other flow. As it stands, we can't just assume that all external tools return one error code for any failure (Gen BE compiler being a good example), or that their single error code matches those from other tools (e.g. `-1` from the SPIR-V Translator).

As far as the `llvm-no-spir-kernel` tool is concerned, the only other known error would occur in case of an invalid input. The driver aims to provide all triggered tools with correct inputs, and the driver LITs are there to check this.

Reply:
I didn't get it. How addition of a `map` helps handling different errors with same error code?

Reply:
> How addition of a map helps handling different errors with same error code?

It doesn't (and I didn't imply it did). It only helps if the tool gives different error codes (which is not unprecedented, as I've pointed out above).

Reply:
Can you document the expectations of unique codes between external tools when using this custom diagnostic interface?  Or even same codes from different tools that have the desire of the same diagnostic from the driver.

Reply:
That's a good idea, will do

Reply:
Done